Creating the json file for each run with an R script

OlintoYale opened the pull request:
This is an R script that can be used to collect data, including the fasta sequences, and save it at Volumes / . . . . .. . /working_json

OlintoYale committed:
first commit

This is an R script that can be used to collect data, including the fasta sequences, and save it at Volumes / . . . . .. . /working_json

erinyoung commented on fasta2JSON.R:
```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env Rscript
+# Author: OLP
+# Usage: Getting metadata for daily_metadata: fast2JOSN file
+# Parameters: Run Name ;  i.e. Rscript fastas2JSON.R UT-VH00770-230212
```
I think there's a typo on this line
```
# Parameter: RunName ; i.e. Rscript fasta2JSON.R UT-VH00770-230212
```

OlintoYale replied:
It has been updated

erinyoung commented on fasta2JSON.R:
```diff
@@ -0,0 +1,128 @@
+  collection_date = Alist_1[[2]],
+  Epitrax_ID = Alist_1[[3]],
+  filename = Alist_1[[4]],
+  sequence = df)
```
`sequence = df` prints strangely to the resultant file

OlintoYale replied:
It as been modify to sprint the results as required

erinyoung commented on fasta2JSON.R:
```diff
@@ -0,0 +1,128 @@
+  filename = Alist_1[[4]],
+  sequence = df)
+
+Path2Save<-paste("",paste("R",args[1],"json",sep="."), sep = "")
```
The filename needs to be run.json. The "R" at the beginning needs to be removed.

OlintoYale replied:
The R has been removed

OlintoYale committed:
Update fasta2JSON.R

The script was modified to be immune to changes in the SampleSheet format, and added a new column related to the lims_found = sampleNumber or missing.

erinyoung commented:
I forgot to ask before, but is there a reason you want to save the json files to `/Volumes/IDGenomics_NAS/COVID/daily_metadata/working_json` instead of `/Volumes/IDGenomics_NAS/COVID/daily_metadata/json`?

OlintoYale commented:
No reason, I just modified the path to json folder

On Fri, Feb 24, 2023 at 2:11 PM Young ***@***.***> wrote:

> I forgot to ask before, but is there a reason you want to save the json
> files to /Volumes/IDGenomics_NAS/COVID/daily_metadata/working_json
> instead of /Volumes/IDGenomics_NAS/COVID/daily_metadata/json?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/UPHL-BioNGS/General_LW_scripts/pull/23#issuecomment-1444497256>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUISDYITCUN5VWCI6UUEYXLWZEPWHANCNFSM6AAAAAAVDSKYX4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
*************** IMPORTANT MESSAGE ***************
This message, including 
any attachments, may contain confidential information intended for a 
specific individual and purpose, and is protected by law.  If you are not 
the intended recipient, delete this message, including from trash, and 
notify me by telephone or email.


If you are not the intended recipient, 
any distributions or copying of this message, or the taking of any action 
based on its content is strictly prohibited.

OlintoYale committed:
Update fasta2JSON.R

OlintoYale committed:
Update fasta2JSON.R

OlintoYale commented:
fasta2JSON.R: All modifications requested have been made

On Thu, Feb 23, 2023 at 10:42 AM Young ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In fasta2JSON.R
> <https://github.com/UPHL-BioNGS/General_LW_scripts/pull/23#discussion_r1116038745>
> :
>
> > +Alist_1 = vector(mode="list", length = counter)
> +Alist_1[[1]] = Matrix$Sample_Accession[1:counter]
> +Alist_1[[2]] = Matrix$collectionDate[1:counter]
> +Alist_1[[3]] = Matrix$patient_record_number[1:counter]
> +Alist_1[[4]] = Matrix$Epitrax_ID[1:counter]
> +Alist_1[[5]] = df
> +
> +
> +json2 <- data.frame(
> +  LIMS_ID = Alist_1[[1]],
> +  collection_date = Alist_1[[2]],
> +  Epitrax_ID = Alist_1[[3]],
> +  filename = Alist_1[[4]],
> +  sequence = df)
> +
> +Path2Save<-paste("",paste("R",args[1],"json",sep="."), sep = "")
>
> The filename needs to be run.json. The "R" at the beginning needs to be
> removed.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/UPHL-BioNGS/General_LW_scripts/pull/23#pullrequestreview-1311824158>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUISDYLU4CLRH3DP7MR5YVDWY6OQJANCNFSM6AAAAAAVDSKYX4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 
*************** IMPORTANT MESSAGE ***************
This message, including 
any attachments, may contain confidential information intended for a 
specific individual and purpose, and is protected by law.  If you are not 
the intended recipient, delete this message, including from trash, and 
notify me by telephone or email.


If you are not the intended recipient, 
any distributions or copying of this message, or the taking of any action 
based on its content is strictly prohibited.

OlintoYale commented:
I closed this PR on July-26-2023. I added the python script (fasta-to-json.py) to be run inside an R Script that generates the ngs/summary files.
